### PR TITLE
Adding docker usage to readme

### LINKED ISSUE #58: [DOCS] Update README.md with Docker usage
Add to the usage section-

## Docker

Replace the `--tf-dir` value with your infra's path, or git clone this repository to check the examples

```
docker run --rm -v $PWD/:/code/ infracost/infracost:latest --tf-dir /code/examples/small_terraform
```

### PR DESCRIPTION
**Pull request by sashman**
Fixes #58

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,6 +43,10 @@ Infracost shows hourly and monthly cost estimates for a Terraform project. This
     Docker:
     ```sh
     docker pull infracost/infracost
+    docker run --rm -e INFRACOST_API_KEY=... \
```
Comment (alikhajeh1) [inline README.md:46]: Does this make it clearer for first time users?

```
docker run --rm \
      -e INFRACOST_API_KEY=see_following_steps_on_how_to_get_this \
      -v $PWD/:/code/ infracost/infracost --tfdir /code/
      # add other required flags for infracost or envs for Terraform
```

Comment (sashman) [reply README.md:46]: I think the AWS ones are handy explicitly as TF doesn't return a very sensible error message in my opinion.

Comment (sashman) [reply README.md:46]: e.g.

> 
>    ✖ Running terraform plan
>   Terraform command failed with:
> 
>     Error: error configuring S3 Backend: no valid credential sources for S3 Backend found.
>     Please see https://www.terraform.io/docs/backends/types/s3.html
>     for more information about providing credentials.
>     Error: NoCredentialProviders: no valid providers in chain. Deprecated.
>     For verbose messaging see aws.Config.CredentialsChainVerboseErrors
> 
> Error running terraform plan: exit status 1

Comment (alikhajeh1) [reply README.md:46]:
```
docker run --rm \
      -e INFRACOST_API_KEY=see_following_steps_on_how_to_get_this \
      -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID \
      -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY \
      -v $PWD/:/code/ infracost/infracost --tfdir /code/
      # add other required flags for infracost or envs for Terraform
```

**Pull request merged**